lib/vkloader.c

```diff
@@ -816,6 +826,14 @@ ktxTexture_VkUploadEx(ktxTexture* This, ktxVulkanDeviceInfo* vdi,
     ktx_uint32_t             numImageLayers, numImageLevels;
     ktx_uint32_t elementSize = ktxTexture_GetElementSize(This);
     ktx_bool_t               canUseFasterPath;
+    // Ensure that all callbacks are provided (or at least not NULL) before we start using them
+    ktx_bool_t               useSuballocator = (subAllocatorCallbacks && 
+                                                subAllocatorCallbacks->allocMemFuncPtr &&
+                                                subAllocatorCallbacks->bindBufferFuncPtr && 
+                                                subAllocatorCallbacks->bindImageFuncPtr && 
+                                                subAllocatorCallbacks->memoryMapFuncPtr && 
+                                                subAllocatorCallbacks->memoryUnmapFuncPtr && 
+                                                subAllocatorCallbacks->freeMemFuncPtr);
 
```

Comment:
I am concerned about silently manually allocate VkDeviceMemory when an effort has been made to provide suballocator callbacks. How about returning KTX_INVALID_OPERATION if subAllocatorCallbacks is non-zero but not all callbacks have been provided.

Reply:
Yea, that works: more explicit.

Reply:
On reflection I think KTX_INVALID_VALUE is better as it is a problem with the value of the subAllocatorCallbacks argument rather than the internal state of the texture object not allowing the operation.

Reply:
Yup, just changed the return value + specs.